### PR TITLE
Commerce 9047 latest 2

### PR DESCRIPTION
**Pull request by MahmoudAzzam**

### CODE REVIEW

**Comment by MahmoudAzzam** [comment]
ci:test:relevant

**Comment by MahmoudAzzam** [comment]
Previous PR : https://github.com/liferay-commerce/liferay-portal/pull/2736

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 1]
```diff
@@ -1741,7 +1742,7 @@ public CPDefinition updateCPDefinition(
 
 		// Commerce product definition
 
-		User user = _userLocalService.getUser(serviceContext.getUserId());
+		User user = _userLocalService.getUser(PrincipalThreadLocal.getUserId());
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:1745]: Ehi @MahmoudAzzam I think this change is not necessary

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 2]
```diff
@@ -1966,7 +1967,7 @@ public CPDefinition updateCPDefinitionCategorization(
 		}
 
 		updateStatus(
-			serviceContext.getUserId(), cpDefinitionId,
+			PrincipalThreadLocal.getUserId(), cpDefinitionId,
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:1970]: also this

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java [hunk 3]
```diff
@@ -1976,7 +1977,7 @@ public CPDefinition updateCPDefinitionCategorization(
 			serviceContext.getCompanyId());
 
 		_assetEntryLocalService.updateEntry(
-			serviceContext.getUserId(), companyGroup.getGroupId(),
+			PrincipalThreadLocal.getUserId(), companyGroup.getGroupId(),
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionLocalServiceImpl.java:1980]: also this

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]
Ehi @MahmoudAzzam the change was in the first place requested by Brian to avoid the use of the service context thread local. In all other places it is fine to get the user id from the service context

**Pull request closed without merging**